src/Swarm/Version.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- |
+-- Module      :  Swarm.Version
+-- Copyright   :  Brent Yorgey
+-- Maintainer  :  byorgey@gmail.com
+--
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Query current and upstream Swarm version.
+module Swarm.Version where
+
+import Control.Monad (forM)
+import Data.Aeson (Array, Value (..), (.:))
+import Data.Aeson.Types (parseMaybe)
+import Data.Bifunctor (first)
+import Data.ByteString qualified as BS
+import Data.ByteString.Lazy qualified as BSL
+import Data.Char (isDigit)
+import Data.Foldable (find, toList)
+import Data.List.Extra (breakOnEnd)
+import Data.Maybe (catMaybes)
+import Data.Version (showVersion)
+import Data.Yaml (ParseException, decodeEither')
+import GitHash (GitInfo, giBranch, giHash, giTag, tGitInfoCwdTry)
+import Network.HTTP.Client (
+  Request (requestHeaders),
+  Response (responseBody),
+  httpLbs,
+  newManager,
+  parseRequest,
+ )
+import Network.HTTP.Client.TLS (tlsManagerSettings)
+import Network.HTTP.Types (hUserAgent)
+import Paths_swarm qualified
+
+gitInfo :: Either String GitInfo
+gitInfo = $$tGitInfoCwdTry
+
+commitInfo :: String
+commitInfo = case gitInfo of
+  Left _ -> ""
+  Right git -> " (" <> giBranch git <> "@" <> take 10 (giHash git) <> ")"
+
+type Hash = String
+
+isSwarmReleaseTag :: String -> Bool
+isSwarmReleaseTag = all (\c -> isDigit c || c == '.')
```

Comment:
I can get anything that has the version as a substring to work. The only requirement is that it should be the same forever, so that old versions of the game can display the notification.

I think using the same format as Cabal with no prefix would make it obvious that the tag name is parsed by machines.
Let's use the `0.1.0.0` format for tags and not be tempted to change prefixes to something prettier.